cmd/comfy/prices_cache.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"errors"
+	"log"
+	"strconv"
+	"time"
+)
+
+type PricesCache struct {
+	prices        map[string]*HourlyPrice
+	fetchCallback FetchPrices
+}
+
+type FetchPrices = func() []HourlyPrice
+
+func initPricesCache(fetchCallback FetchPrices) *PricesCache {
+	var pricesCache = &PricesCache{prices: nil, fetchCallback: fetchCallback}
+
+	pricesCache.refreshPrices(pricesCache.fetchCallback())
+
+	return pricesCache
+}
+
+func (pricesCache *PricesCache) refreshPrices(pricesToday []HourlyPrice) {
+	log.Printf("Refreshing prices!")
+	pricesCache.prices = make(map[string]*HourlyPrice, 48)
+	for _, value := range pricesToday {
+		var key string = pricesCache.getLookupKey(value.StartsAt)

```

Comment:
maybe we can provide entire value and `getLookupKey` can decide internally what is needed? This decreases the mental overhead when reading this function

Reply:
since it's called with only the time available when fetching price, let's leave it for now